src/components/EditCoursePage/index.jsx

```diff
@@ -370,6 +370,7 @@ class EditCoursePage extends React.Component {
       organization_logo_override: courseData.organization_logo_override_url,
       organization_short_code_override:
         courseData.organization_short_code_override,
+      watchers: courseData.watchers_list ? courseData.watchers_list.map((watcher) => watcher.value) : [],
```

Comment:
wouldn't this turn watchers into empty list if a non-admin user (like Editor) publishes or edits the course?

Reply:
Try opening the course with an Editor user. I suspect the above will be true.

Reply:
nope, it doesn't turn into an empty list I've already tried this for course Editor Role on my local. And we are not showing this field to Course Editor. Only Admin user/ PCs can view or edit this field.

#### Course Editor View:
<img width="934" alt="image" src="https://github.com/openedx/frontend-app-publisher/assets/78806673/205f2f40-18a2-499c-949e-6f7542167af7">


#### Admin User/ PC view:
<img width="937" alt="image" src="https://github.com/openedx/frontend-app-publisher/assets/78806673/bc6faf55-84d4-4163-9de6-867bfda53faa">

I've tried updating the course with course Editor role user it didn't empty the watchers field

Reply:
But we are adding key in the request data and are adding [] if the form data is not found. What's the request value in network tab when editing as an Editor user? Can you double check if we can remove watchers from a course (assuming that if [] is not changing watchers on backend, then we cannot remove all watchers for a course when needed).

Reply:
Actually form data initialValues send to `prepareSendCourseData` method if we don't show change them, as Watchers field is not showing with `Couse Editor` role on Course Edit Form so they can't edit this field.  `prepareSendCourseData` doesn't send empty watchers field to backend because we're not editing watchers initial values. It send same previous watcher value to backend.

> (assuming that if [] is not changing watchers on backend, then we cannot remove all watchers for a course when needed).
We can empty this field when needed by editing this field with PCs/Admin Role. `[]` this is only possible when watchers value is `null` not in case of Course Editor editing the form because in that case it's sending previous value to backend

Reply:
- https://github.com/openedx/frontend-app-publisher/pull/890/files#diff-1018929192344c42ef49da293a7c2bd439019169f1fa3bc10b84b6b2805fe999R666
Here we are setting the initial value of `watchers` coming from backend in format for multi email field

- https://github.com/openedx/frontend-app-publisher/pull/890/files#diff-1018929192344c42ef49da293a7c2bd439019169f1fa3bc10b84b6b2805fe999R666
Here we are passing this formatted value to the field (if we don't show this field its state remain unaffected)

- https://github.com/openedx/frontend-app-publisher/pull/890/files#diff-1018929192344c42ef49da293a7c2bd439019169f1fa3bc10b84b6b2805fe999R373
Here we're converting watchers back to list format

(If we hide the "watchers" field for course editors, it will not affect the "watchers_list" state. During the onSave event, the code will retain the existing state of "watchers_list" rather than passing an empty state)